### PR TITLE
[SPARK-30066][SQL][FOLLOWUP] Remove size field for interval column cache

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

A followup for #26699, clear the size field for interval column cache, which is needless and can reduce the memory cost.

### Why are the changes needed?
followup

### Does this PR introduce any user-facing change?

no

### How was this patch tested?

existing ut.

### CODE REVIEW

**Comment by yaooqinn** [comment]
@cloud-fan @dongjoon-hyun @maropu, thanks all for take a look at this followup. And excuse me for bringing extra works to you guys.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnType.scala [hunk 1]
```diff
@@ -705,14 +705,11 @@ private[columnar] case class MAP(dataType: MapType)
...
-private[columnar] object CALENDAR_INTERVAL extends ColumnType[CalendarInterval]
-  with DirectCopyColumnType[CalendarInterval] {
+private[columnar] object CALENDAR_INTERVAL extends ColumnType[CalendarInterval] {
 
   override def dataType: DataType = CalendarIntervalType
 
-  override def defaultSize: Int = 16
-
-  override def actualSize(row: InternalRow, ordinal: Int): Int = 20
+  override val defaultSize: Int = 16
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnType.scala]: is it better to use `def`? to follow `def dataType`.

Comment (yaooqinn) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnType.scala]: ok

File: sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnType.scala [hunk 2]
```diff
@@ -705,31 +705,35 @@ private[columnar] case class MAP(dataType: MapType)
...
-    ByteBufferHelper.getInt(buffer)
     val months = ByteBufferHelper.getInt(buffer)
     val days = ByteBufferHelper.getInt(buffer)
     val microseconds = ByteBufferHelper.getLong(buffer)
     new CalendarInterval(months, days, microseconds)
   }
 
+  // copy the bytes from ByteBuffer to UnsafeRow
+  override def extract(buffer: ByteBuffer, row: InternalRow, ordinal: Int): Unit = row match {
+    case r: MutableUnsafeRow =>
```
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnType.scala]: nit: How about just using `if` in the same way with decimal?: https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnType.scala#L478-L484

**Review by maropu — APPROVED** [review]
LGTM except for one minor sytle comment.

**Review by cloud-fan — APPROVED** [review]

**Comment by maropu** [comment]
Thanks! Merged to master.

**Comment by yaooqinn** [comment]
thanks for merging and reviewing.

**Pull request closed without merging**